src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -28,8 +29,14 @@ internal static class CodeRefactoringHelpers
         /// of tokens gracefully.
         /// </para>
         /// </summary>
-        public static async Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(
+        public static Task<TSyntaxNode> TryGetSelectedNodeAsync<TSyntaxNode>(
             Document document, TextSpan selection, CancellationToken cancellationToken) where TSyntaxNode : SyntaxNode
+        {
+            return TryGetSelectedNodeAsync<TSyntaxNode>(document, selection, (n) => n, cancellationToken);
```

Comment:
Isn't there some cached identity `Func<>` already created?

Reply:
`n => n` will cache automatically since it doesn't capture anything.  You can also use Functions.Identity.